src/System.Net.Http/src/System/Net/Http/Managed/AuthenticationHelper.Digest.cs

```diff
@@ -251,128 +244,145 @@ private static string ComputeHash(string data, string algorithm)
             }
         }
 
-        public class DigestResponse
+        internal class DigestResponse
         {
-            public readonly Dictionary<string, string> Parameters = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+            internal readonly Dictionary<string, string> Parameters = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
 
             // Keep track of request values for this response.
-            public string Nonce;
-            public uint NonceCount;
+            internal const int NonceCount = 1;
 
-            public DigestResponse(string challenge)
+            internal DigestResponse(string challenge)
             {
-                Nonce = null;
-                NonceCount = 1;
-
                 Parse(challenge);
             }
 
-            private unsafe string GetNextKey(ref char* p)
+            private string GetNextKey(string data, int currentIndex, out int parsedIndex)
             {
-                // It is generally cheaper to change a local and then write back to ref at the end
-                // rather than updating the ref on each operation.
-                char* temp = p;
-
-                StringBuilder sb = StringBuilderCache.Acquire();
-
                 // Skip leading whitespace
```

Comment:
Is ' ' the only whitespace that needs to be considered?  Generally when we say "whitespace", that includes things like '\n', '\t', etc. as well.  Either the code or the comment should be clarified.

Reply:
HTAB = '\t' is also counted as whitespace in RFC.